pkg/archive/diff.go

```diff
@@ -224,6 +224,25 @@ func ApplyUncompressedLayer(dest string, layer io.Reader, options *TarOptions) (
 	return applyLayerHandler(dest, layer, options, false)
 }
 
+// IsEmpty checks if the tar archive is empty (doesn't contain any entries).
+func IsEmpty(rd io.Reader) (bool, error) {
```

Comment:
And now I'm wondering; could we still detect this using "Ye olde sha" of the empty tar? 

https://github.com/moby/moby/pull/8827

> Goodbye '511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158',
> it was nice knowing you.

Reply:
Added a fast path.

Reply:
^^ for future reference; this was reverted; it turned out that there was no canonical digest for "empty archive", and the code below was not "too heavy", so it wouldn't gain us much.